### ISSUE TITLE
steps.<id>.outputs.json not found for Docker Metadata Action

### ISSUE THREAD

**Issue by janderssonse** [rhysd/actionlint#132]
When using the official https://github.com/marketplace/actions/docker-metadata-action as per example. i.e like this: (taken from example in action)
```yaml
 .....
        name: Login to DockerHub
        if: github.event_name != 'pull_request'
        uses: docker/login-action@v1
        with:
          username: ${{ secrets.DOCKERHUB_USERNAME }}
          password: ${{ secrets.DOCKERHUB_TOKEN }}
      -
        name: Build and push
        uses: docker/build-push-action@v2
        with:
          context: .
          push: ${{ github.event_name != 'pull_request' }}
          tags: ${{ steps.meta.outputs.tags }}
          labels: ${{ steps.meta.outputs.labels }}
          build-args: |
            BUILDTIME=${{ fromJSON(steps.meta.outputs.json).labels['org.opencontainers.image.created'] }}
            VERSION=${{ fromJSON(steps.meta.outputs.json).labels['org.opencontainers.image.version'] }}
            REVISION=${{ fromJSON(steps.meta.outputs.json).labels['org.opencontainers.image.revision'] }}
```

The output (steps.meta.outputs.json) gets an "property json is not defined", although this way is the working, recommended syntax, see https://github.com/marketplace/actions/docker-metadata-action#json-output-object

**Comment by rhysd** [comment]
I tried the example workflow in the action.

```yaml
name: ci

on:
  push:
    branches:
      - 'master'
    tags:
      - 'v*'
  pull_request:
    branches:
      - 'master'

jobs:
  docker:
    runs-on: ubuntu-latest
    steps:
      -
        name: Checkout
        uses: actions/checkout@v2
      -
        name: Docker meta
        id: meta
        uses: docker/metadata-action@v3
        with:
          images: name/app
      -
        name: Login to DockerHub
        if: github.event_name != 'pull_request'
        uses: docker/login-action@v1
        with:
          username: ${{ secrets.DOCKERHUB_USERNAME }}
          password: ${{ secrets.DOCKERHUB_TOKEN }}
      -
        name: Build and push
        uses: docker/build-push-action@v2
        with:
          context: .
          push: ${{ github.event_name != 'pull_request' }}
          tags: ${{ steps.meta.outputs.tags }}
          labels: ${{ steps.meta.outputs.labels }}
```

And actionlint didn't report any error.

**Comment by janderssonse** [comment]
Thanks for that but the example you tested does not use the false positive with  steps.meta.outputs.json. You would have to try the specific example here: https://github.com/marketplace/actions/docker-metadata-action#json-output-object (which is using the steps.meta.outputs.json). Just add the build-args: part from that example  to the end of your example above and you will be able reproduce it.

**Comment by rhysd** [comment]
@janderssonse Can you share the entire workflow to reproduce this issue?

**Comment by rhysd** [comment]
The example is a snippet, not an entire workflow.

I tried to make a minimal workflow based on the example, but it did not reproduce any error.

```yaml
on: push

jobs:
  test:
    runs-on: ubuntu-latest
    steps:
      - name: Docker meta
        uses: docker/metadata-action@v3
        id: meta
        with:
          images: name/app
      - name: Build and push
        uses: docker/build-push-action@v2
        with:
          tags: ${{ steps.meta.outputs.tags }}
          labels: ${{ steps.meta.outputs.labels }}
          build-args: |
            BUILDTIME=${{ fromJSON(steps.meta.outputs.json).labels['org.opencontainers.image.created'] }}
            VERSION=${{ fromJSON(steps.meta.outputs.json).labels['org.opencontainers.image.version'] }}
            REVISION=${{ fromJSON(steps.meta.outputs.json).labels['org.opencontainers.image.revision'] }}
```

**Comment by rhysd** [comment]
@janderssonse I tried the workflow but it did not report any error. Can you show output of `actionlint -version`?

**Comment by janderssonse** [comment]
1.6.11 - i'm using it through megalinter. So, maybe it got fixed for 1.6.12? But copy paste into the online linter gives the same error. Is it using an version < 1.6.12?

**Comment by rhysd** [comment]
1.6.11 seems too old. I downloaded 1.6.11 binary and it reproduced this issue. And 1.6.12 did not reproduce it. Please update actionlint to 1.6.12.
